### PR TITLE
[#12048] SQL injection test for UsersDbIT 

### PR DESCRIPTION
**Pull request by cedricongjh**
Part of #12048

**Outline of Solution**
Added sql injection tests to relevant methods in UsersDb

### CODE REVIEW

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java [hunk 1]
```diff
@@ -44,6 +47,12 @@ public void setUp() throws Exception {
         course = new Course("course-id", "course-name", Const.DEFAULT_TIME_ZONE, "institute");
         coursesDb.createCourse(course);
 
+        section = new Section(course, "test-section");
+        course.addSection(section);
+        team = new Team(section, "test-section");
```
Comment (weiquu) [inline src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java:52]: Let's make the team name `"test-team"` to avoid any potential future confusion

**Review by weiquu — COMMENTED** [review]
small nit, otherwise lgtm

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java (hunk 1 above)
Comment (cedricongjh) [reply src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java]: changed!

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java [hunk 2]
```diff
@@ -276,4 +286,301 @@ public void testGetStudentsByGoogleId()
...
+    }
+
+    @Test
+    public void testSqlInjectionInGetStudentByRegKey() throws Exception {
+        ______TS("SQL Injection test in getStudentByRegKey");
+
+        Student student = getTypicalStudent();
+        student.setEmail("studentregkey.usersdbit@gmail.com");
+
+        usersDb.createInstructor(instructor);
```
Comment (jayasting98) [inline src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java:403]: It seems unnecessary to create an instructor here.

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java [hunk 3]
```diff
@@ -276,4 +286,301 @@ public void testGetStudentsByGoogleId()
...
+
+    @Test
+    public void testSqlInjectionInGetTeam() throws Exception {
+        String injection = "test' OR 1 = 1; --";
+        ______TS("SQL Injection test in getTeam courseId field");
+        assertNull(usersDb.getSection(injection, team.getName()));
+
+        ______TS("SQL Injection test in getTeam teamName field");
+        assertNull(usersDb.getSection(course.getId(), injection));
+    }
```
Comment (jayasting98) [inline src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java:585]: I think you forgot to change the method when copy-pasting. This doesn't test `getTeam`.

**Review by jayasting98 — CHANGES_REQUESTED** [review]
I think you may have missed `getTeamOrCreate` and `updateStudent`. Or were they left out for a reason?

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java (hunk 2 above)
Comment (cedricongjh) [reply src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java]: ah oops, I was meant to create a student here, changed

File: src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java (hunk 3 above)
Comment (cedricongjh) [reply src/it/java/teammates/it/storage/sqlapi/UsersDbIT.java]: good catch, changed!

**Comment by cedricongjh** [comment]
> I think you may have missed `getTeamOrCreate` and `updateStudent`. Or were they left out for a reason?

that's my bad, i've added the tests for those methods

**Review by jayasting98 — APPROVED** [review]
Looks good now. Thanks!

**Comment by weiquu** [comment]
@cedricongjh some lint checks failing

**Review by weiquu — APPROVED** [review]
LGTM

**Pull request merged**